Videos/Smart Configurator Videos_RX.htm

```diff
@@ -17,71 +18,254 @@
-                    var description = elem.getElementsByTagName("description")[0].textContent;
-                    var date = elem.getElementsByTagName("date")[0].textContent;
-                    var link = elem.getElementsByTagName("renesaslink")[0].textContent;
-
-                    var recent = elem.getAttribute("recent");
-                    if (new_video == 0 && recent == "true") {
-                        document.getElementById("banner").innerHTML += "<div class=\"info notification\">" +
-                            "<span aria-hidden=\"true\" class=\"material-icons\">info</span><strong>New Videos</strong>" +
-                            "<ul id=\"new-video\" style=\"margin-bottom: 0.2em;\">" +
-                                "<li style=\"list-style-type: none;\">" + date + 
-                                    ": <a href=\"" + link + "\">" + title + "</a></li></ul></div>";
-                        new_video++;
-                    } else if (recent == "true") {
-                        document.getElementById("new-video").innerHTML += "<li style=\"list-style-type: none;\">" + date + 
-                            ": <a href=\"" + link + "\">" + title + "</a></li>";
+
+                    var cloneData = infoData.cloneNode(true); // duplicate infoData to prevent data corruption
+                    filteredXmlDoc.appendChild(cloneData);
+                    // append new videos(within monthRange) or latest 5 videos (whichever is bigger)
+                    if (isNewVideo(infoData)) {
+                        insertToNewVidList(cloneData);
+                    }
+                }
+                if (newVideoArr.length < 5) {
+                    for (var i = 0; i < latestFiveArr.length; i++) {
+                        var cloneFrId = xmlDoc.getElementById(latestFiveArr[i].strId);
+                        filteredNewVideoDoc.appendChild(cloneFrId);
+                    }
+                } else {
+                    for (var i = 0; i < newVideoArr.length; i++) {
+                        var cloneFrId = xmlDoc.getElementById(newVideoArr[i].strDataId).cloneNode(true);
+                        filteredNewVideoDoc.appendChild(cloneFrId);
                     }
+                }
+
+                // convert to HTMLCollection
+                filteredXmlList = filteredXmlDoc.getElementsByTagName("info");
+                filteredNewVideoList = filteredNewVideoDoc.getElementsByTagName("info");
+            }
 
-                    document.getElementById("video-wrapper").innerHTML += "<div id=\"" + id + "\" class=\"video-contents\">" + 
-                      "<iframe src=\"" + video + "\" allowfullscreen=allowfullscreen></iframe><h3>" + title + "</h3>" + 
-                      "<div class=\"video-description\"><p>" + description + "</p><p><br>Date of upload: " + date + "</p></div>" + 
-                      "<a href=\"" + link + "\">View on Renesas website</a></div>";
+            function isNewVideo(infoElement) {
```

Comment:
Q: Isn't this code duplicated into each HTML? (not sure if there is any way to share the <script> element between HTML?

Reply:
This function, isNewVideo(), is duplicated accross the files but it also manipulates global variables, similar for insertToNewVidList().
I think it is possible to make the entire script a common script file for all but it will require more changes as these script have device specific conditions for each file....